projects/plugins/jetpack/_inc/client/recommendations/timer/index.jsx

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import React, { useState, useEffect, useCallback, useRef, useMemo } from 'react';
+
+const Timer = ( { label, timeClassName, expiryDate } ) => {
+	const date = useMemo( () => new Date( expiryDate ), [ expiryDate ] );
+	const [ duration, setDuration ] = useState( date - Date.now() );
+	const intervalRef = useRef();
+
+	const tick = useCallback( () => {
+		setDuration( date - Date.now() );
+	}, [ date, setDuration ] );
+
+	const format = useCallback( ts => {
+		const d = ts / 1000;
+		const days = Math.floor( d / ( 3600 * 24 ) );
+		const hours = Math.floor( ( d - days * 3600 * 24 ) / 3600 );
+		const minutes = Math.floor( ( d - days * 3600 * 24 - hours * 3600 ) / 60 );
+		const seconds = Math.floor( d - days * 3600 * 24 - hours * 3600 - minutes * 60 );
+		const parts = [];
+
+		if ( days > 0 ) {
+			// translators: %d is the number of days, d an abbreviation for days (e.g. 7d). Only translate the latter.
+			parts.push( sprintf( __( '%dd', 'jetpack' ), days ) );
+		}
+
+		if ( days > 0 || hours > 0 ) {
+			// translators: %d is the number of hours, h an abbreviation for hours (e.g. 23h). Only translate the latter.
+			parts.push( sprintf( __( '%dh', 'jetpack' ), hours ) );
+		}
+
+		if ( days > 0 || hours > 0 || minutes > 0 ) {
+			// translators: %d is the number of minutes, m an abbreviation for minutes (e.g. 59m). Only translate the latter.
+			parts.push( sprintf( __( '%dm', 'jetpack' ), minutes ) );
+		}
+
+		// translators: %d is the number of seconds, s an abbreviation for seconds (e.g. 59s). Only translate the latter.
+		parts.push( sprintf( __( '%ds', 'jetpack' ), seconds ) );
+
+		return parts.join( ' ' );
```

Comment:
I worry this is going to be hard to translate, since not every language handle numbers the same (especially in regards to singular / plural, which we do not support here), and not every language expects date formats to be elements separated by spaces like this. I wonder if we could make it one single translatable string with multiple elements instead, or take an approach where we would only display the most important values instead, kinda like `human_time_diff` does ([reference](https://developer.wordpress.org/reference/functions/human_time_diff/)). 

It's a pity we don't have it available in js yet 🙏 
https://github.com/WordPress/gutenberg/issues/14486

Reply:
It seems to me that this format (first letter of the unit, and spaces between elements) is imposed by the design and the relative lack of space. I don't think the idea was to display a human readable time difference, as we usually see it.
<img width="268" alt="Screen Shot 2022-04-08 at 9 53 36 AM" src="https://user-images.githubusercontent.com/1620183/162451191-5680ac22-4c87-431b-a289-ac77bf6b5d75.png">

With that in mind, I don't see any case in which that'd be an issue. Do you have an example, maybe?

Reply:
I only speak French and Hungarian, where singular and plural would be handled the same way here so I don't have a great example myself I'm afraid. It may be worth asking for someone from @Automattic/i18n to review, they'll know better than anyone. But in any case, I think that's something that can be handled in a follow-up PR if changes need to be made.

Reply:
Gotcha. Thanks for pointing it out. I'll follow up with them and create a new PR if necessary.

Reply:
@monsieur-z 
The pluralization and order issues were already discussed in Slack, so just adding my notes. In most locales the single letter abbreviation would work, but in some locales the full word had to be used (Arabic, Hebrew, Korean etc.), which might not fit if there is a limited amount of space. I also see that some translators just preserved the original strings. A simple solution would be to use the `%02d:%02d:%02d:%02d` format and not translate it. Another approach would be to create four translatable strings (one for just seconds, one for hours and seconds and so on), mention if there is a character limit and let the translators to decide whether to use abbreviations or dd:hh:mm:ss, but the danger here is that it might not be consistent and after seeing `01:00` the user would be shown `59 seconds`.